File: core/src/com/mygdx/game/deckengine/card/StrikeCard.java
package com.mygdx.game.deckengine.card;

import com.badlogic.gdx.graphics.Texture;
import com.mygdx.game.character.CharacterStats;
import com.mygdx.game.character.CharacterType;

public class StrikeCard extends Card {

    public StrikeCard(CardLayout cardLayout, Texture img) {
        super(cardLayout, img);
    }

    @Override
    public int GetEnergyCost() {
        return 1;
    }

    @Override
    public boolean CanApplyEffects(CharacterStats stats) {

        return (stats.getCharacterType() == CharacterType.ENEMY);
    }

    @Override
    public void ApplyEffects(CharacterStats stats) {
        stats.AdjustHealth(-6);
    }
}

File: combat/src/classicthunder/components/card/StrikeCard.kt
package classicthunder.components.card

import classicthunder.components.character.CharacterStats
import classicthunder.components.character.CharacterType
import com.badlogic.gdx.graphics.Texture

class StrikeCard(texture: Texture) : Card(texture, 1) {
    override fun GetEnergyCost(): Int {
        return 1
    }

    override fun CanApplyEffects(stats: CharacterStats): Boolean {
        return stats.characterType == CharacterType.ENEMY
    }

    override fun ApplyEffects(stats: CharacterStats) {
        stats.AdjustHealth(-6)
    }
}